Source/DafnyCore/Coco/Parser.frame

```diff
@@ -167,6 +173,13 @@ public class Errors {
     return s;
   }
 
+  public void SemErr(string errorId, IToken tok, string msg) {  // semantic errors
```

Comment:
Doesn't this also mean you always have errorIds ? The string parameter is not optional

Reply:
Plugins will use `Reporting.Error` directly, they won't use the internal method `SemErr`. For our own code, every method should get an error id so that it can be linked to documentation, and also possibly used by code actions.

Reply:
Even though they can't now, plugins should be able to add syntax as well. I don't think we should have two almost the same but slightly different variations of an API, one for plugins and one for code from this repo.